pkg/k8sutil/rollout.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2021 The Lokomotive Authors
+// Copyright 2016 The Kubernetes Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sutil
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientappsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+)
+
+// RolloutTimeout is the timeout to wait for a daemonset or deployment to rollout changes.
+const RolloutTimeout = 15 * time.Minute
+
+// RolloutRestartDaemonSet is the programmatic equivalent of 'kubectl rollout restart'.
+func RolloutRestartDaemonSet(dsi clientappsv1.DaemonSetInterface, name string) (int64, error) {
+	ds, err := dsi.Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
We should be accepting context as a parameter to this and other functions added to this package.